Message Board/Model Controller/MessageThreadController.swift

```diff
@@ -31,7 +31,24 @@ class MessageThreadController {
             guard let data = data else { NSLog("No data returned from data task"); completion(); return }
             
             do {
-                self.messageThreads = try JSONDecoder().decode([MessageThread].self, from: data)
+                let results = try JSONDecoder().decode([String : MessageThread].self, from: data)
+                let resultsArray = Array(results.values)
+                
+                var sorted: [MessageThread] = []
+                
+                for thread in resultsArray {
```

Comment:
I loved that you fixed message ordering. 

There are bugs that are obvious, and then there are some that require an understanding of how an app should work.

It would be nice to see a test that verifies this fix actually works (using your test data, add more data points to verify the ordering)

Or you can extract the logic into a helper method, so that you can unit test it outside the UI. Both tests are useful, as they help readers understand the rules for how the app should work.